pkg/chaosdaemon/stress_server.go

```diff
@@ -0,0 +1,241 @@
+		cgroup, _ = filepath.Split(cgroup)
+	}
+	control, err := cgroups.Load(cgroups.V1, cgroups.StaticPath(cgroup))
+	if err != nil {
+		return nil, err
+	}
+	cmd := exec.Command("stress-ng", strings.Fields(req.Stressors)...)
+	if err := cmd.Start(); err != nil {
+		return nil, err
+	}
+	procState, err := process.NewProcess(int32(cmd.Process.Pid))
+	if err != nil {
+		return nil, err
+	}
+	ct, err := procState.CreateTime()
+	if err != nil {
+		if kerr := cmd.Process.Kill(); kerr != nil {
+			log.Error(kerr, "kill stressors failed", "request", req)
+		}
+		return nil, err
+	}
+	if err = control.Add(cgroups.Process{Pid: cmd.Process.Pid}); err != nil {
+		if kerr := cmd.Process.Kill(); kerr != nil {
+			log.Error(kerr, "kill stressors failed", "request", req)
+		}
+		return nil, err
+	}
+	go func() {
+		if err, ok := cmd.Wait().(*exec.ExitError); ok {
+			status := err.Sys().(syscall.WaitStatus)
+			if status.Signaled() && status.Signal() == syscall.SIGKILL {
+				log.Info("Stressors cancelled", "request", req)
+			} else {
+				log.Error(err, "stressors exited accidentally", "request", req)
+			}
+		}
+	}()
+
+	return &pb.ExecStressResponse{
+		Instance:  strconv.Itoa(cmd.Process.Pid),
+		StartTime: ct,
+	}, nil
+}
+
+func (s *daemonServer) CancelStressors(ctx context.Context,
+	req *pb.CancelStressRequest) (*empty.Empty, error) {
+	pid, err := strconv.Atoi(req.Instance)
+	if err != nil {
+		return nil, err
+	}
+	log.Info("Canceling stressors", "request", req)
+	ins, err := process.NewProcess(int32(pid))
+	if err != nil {
+		return &empty.Empty{}, nil
+	}
+	if ct, err := ins.CreateTime(); err == nil && ct == req.StartTime {
+		if err := ins.Kill(); err != nil {
+			return nil, err
+		}
+	}
+	return &empty.Empty{}, nil
+}
+
+func findValidCgroup(path cgroups.Path, target string) (string, error) {
+	for _, subsys := range cgroupSubsys {
+		if p, _ := path(cgroups.Name(subsys)); strings.Contains(p, target) {
+			return p, nil
+		}
+	}
+	return "", fmt.Errorf("never found valid cgroup for %s", target)
+}
+
+// pidPath will return the correct cgroup paths for an existing process running inside a cgroup
+// This is commonly used for the Load function to restore an existing container.
+//
+// Note: it is migrated from cgroups.pidPath since it will find mountinfo incorrectly inside
+// the daemonset. Hope we can fix it in official cgroups repo to solve it.
+func pidPath(pid int) cgroups.Path {
+	p := fmt.Sprintf("/proc/%d/cgroup", pid)
+	paths, err := parseCgroupFile(p)
+	if err != nil {
+		return errorPath(errors.Wrapf(err, "parse cgroup file %s", p))
+	}
+	return existingPath(paths, pid, "")
+}
+
+func errorPath(err error) cgroups.Path {
+	return func(_ cgroups.Name) (string, error) {
+		return "", err
+	}
+}
+
+func existingPath(paths map[string]string, pid int, suffix string) cgroups.Path {
+	// localize the paths based on the root mount dest for nested cgroups
+	for n, p := range paths {
+		dest, err := getCgroupDestination(pid, string(n))
```

Comment:
Why not just read `/proc/pid/mountinfo` out of the loop, then just check `n` inside the loop? WDYT @AngleNet ?

Reply:
I think that also works. I just steal the code from `github.com/containerd/cgroups` package since the official version could not find mountinfo correctly. Hope it will fixed and we could remove the entire stealling pieces and use the official one.